application/client/src/app/ui/views/sidebar/search/filters/provider.ts

```diff
@@ -172,15 +175,34 @@ export class ProviderFilters extends Provider<FilterRequest> {
         if (selected.length !== 1) {
             return [];
         }
-        const entity = selected[0].extract();
+        const entity = selected[0];
+        const request = selected[0].extract();
         const items: IMenuItem[] = [];
-        if (entity instanceof FilterRequest) {
+        if (request instanceof FilterRequest) {
             items.push({
                 caption: `Show Matches`,
                 handler: () => {
-                    this.search(selected[0]);
+                    this.search(entity);
                 },
             });
+            ChartRequest.isValid(request.definition.filter.filter) &&
+                items.push({
+                    caption: `Convert to Chart`,
+                    handler: () => {
+                        this.session.search.store().filters().delete([request.uuid()]);
+                        this.session.search
+                            .store()
+                            .charts()
+                            .update([
+                                new ChartRequest({
```

Comment:
Update method gets `StoredEntity<T>`, but you are sending `ChartRequest`. That's mistake